src/backend/binaryen/wasm_expr_gen.ts

```diff
@@ -1923,108 +1942,176 @@ export class WASMExpressionGen {
         rightValue?: SemanticsValue,
     ) {
         const owner = value.owner as VarValue;
-        const meta = owner.shape!.meta;
-        const member = meta.members[value.index];
+        const shapeMeta = owner.shape!.meta;
+        const shapeMember = shapeMeta.members[value.index];
         const ownerType = owner.type as ObjectType;
         let targetValue = value.value!;
         if (rightValue) {
             targetValue = rightValue;
         }
         const typeMeta = ownerType.meta;
-        return this.setInstField(
+        const typeMember = typeMeta.findMember(
+            shapeMember.name,
+        ) as MemberDescription;
+        return this.setInstMember(
             this.wasmExprGen(owner),
             targetValue,
             ownerType,
             typeMeta,
-            member,
+            typeMember,
         );
     }
 
-    private setInstField(
+    private setInstMember(
```

Comment:
We have `field`, `property` and here `member`, what's the different between these concepts?

Reply:
`field` is just for object's field, method is not contained. 
`property` contains all properties, including field, method. 
In here, `member` is just like `property`, maybe I should rename `xxMember` to `xxProperty` for misunderstanding.

Reply:
xxProperty